assets/js/dashboardModal.js

```diff
@@ -1,33 +1,53 @@
-import {$} from './fquery';
+import {$, all} from './fquery';
+
+const findNodeWith = (element, expectedAttribute) => {
+  while(element) {
+    if (element.getAttribute(expectedAttribute)) {
+      break;
+    }
+    element = element.parentElement;
+  }
+
+  return element
+}
+
+const toggle = e => {
+  $(".modal").toggle("is-active");
+  $("body").toggle("is-active");
+};
 
 export const modal = {
   bind: selector => {
-    const toggle = e => {
-      $(".modal").toggle("is-active");
-      $("body").toggle("is-active");
-    };
 
-    $(selector).on("click", toggle);
+    all(selector).forEach((node) => {
+      node.on('click', (e) => {
+        let element = findNodeWith(e.target, "data-questionnaire-id")
+
+        const questionnaireId = element.getAttribute("data-questionnaire-id");
+        const mentee = element.getAttribute("data-mentee-name");
+
+        $('#modal-title').text = `Confirm for ${mentee}`
+
+        toggle()
+
+        $(".remove").on("click", () => {
```

Comment:
Are you using JQuery only for the selector?

Reply:
Its fQuery!